packages/ui/src/composables/useThrottle.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @description returns throttled function or value
+ * the last one always returns last-call value in the end if no more new calls were provided
+ * @example
+ *    import { useThrottleFunction, useThrottleValue } from '../../composables'
+ *    ...
+ *    const localThrottledFunction = useThrottleFunction(functionToThrottle, props)
+ *    const localThrottledValue = useThrottleValue(reactiveValueToThrottle, props)
+ */
+
+import {
+  ref, toRef, unref,
+  watch,
+  Ref, ExtractPropTypes, ComponentInternalInstance,
+} from 'vue'
+
+type ThrottledFunctionArgs = any[]
+type ThrottledFunction<Output> = (...args: ThrottledFunctionArgs) => Output
+
+export const useThrottleProps = {
+  delay: {
+    type: Number,
+    default: 0,
+    validator: (value: number) => value >= 0,
+  },
+}
+
+type UseThrottleProps = ExtractPropTypes<typeof useThrottleProps>
+
+/**
+ * @param fn passed function to throttle
+ * @param props { delay } call delay in ms
+ */
+export function useThrottleFunction<Output> (fn: ThrottledFunction<Output>, props: UseThrottleProps) {
+  const delay = toRef(props, 'delay') ?? 0
+
+  const isThrottled = ref(true)
+
+  /**
+   * No way this will be returned without reassignment, so we don't want typescript
+   * to always keep undefined as possible return type. If function returns undefined itself
+   * it will be still presented by typescript as undefined (expected behaviour).
+   */
+  const lastCallResult = ref<Output>(undefined as any as Output) as Ref<Output>
+
+  return function (this: ComponentInternalInstance, ...args: ThrottledFunctionArgs) {
```

Comment:
Can't see any reason why here `this` is a `ComponentInternalInstance`. For example, in `useSortable` there is no this in function. BTW, there likely always no this, cuz it used in setup function.

If there will be a chance to have this, you can always use bind().

Reply:
I'll change the type to any, but prefer to leave it:
- i'm not absolutely sure that there're no way to have no `this` - current state serves both versions;
- we should pass `this` anyway, even it's an empty object.